include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -178,28 +179,73 @@ struct operational_domain
      * different physical parameters, the parameters are stored with the corresponding operating status.
      */
     locked_parallel_flat_hash_map<Key, Value> operational_values{};
+    /**
+     * This can store different information depending on the use case. If the critical temperature is simulated on top
+     * of the operational domain, it is stored here.
+     */
+    locked_parallel_flat_hash_map<Key, MetricType> metric_values{};
     /**
      * This function retrieves the value associated with the provided key from the operational domain. If
      * the key is found in the domain, its corresponding value is returned. Otherwise, `std::nullopt`
      * is returned.
      *
      * @param key The key to look up.
-     * @return The value associated with the parameter point.
+     * @return The value associated with the provided key. If the key is not found, `std::nullopt` is returned.
      */
     [[nodiscard]] std::optional<Value> get_value(const Key& key) const
     {
         return detail::contains_key(operational_values, key);
     }
-
+    /**
+     * This function retrieves the value associated with the provided key from the metric values. If
+     * the key is found, its corresponding value is returned. Otherwise, `std::nullopt`
+     * is returned.
+     *
+     * @param key The key to look up in the metric values.
+     * @return The value associated with the provided key. If the key is not found, `std::nullopt` is returned.
+     */
+    [[nodiscard]] std::optional<MetricType> get_metric_value(const Key& key) const
+    {
+        return detail::contains_key(metric_values, key);
+    }
+    /**
+     * Adds a value to the operational domain.
+     *
+     * @key The key to add the value to.
+     * @value The value to add.
+     */
     void add_value(const Key& key, const Value& value)
     {
         operational_values.try_emplace(key, value);
     }
-
-    [[nodiscard]] locked_parallel_flat_hash_map<Key, Value> get_domain() const
+    /**
+     * Adds a value to the metric values.
+     *
+     * @param key The key to add the value to.
+     * @param value The value to add.
+     */
+    void add_metric_value(const Key& key, const MetricType& value)
+    {
+        metric_values.try_emplace(key, value);
+    }
+    /**
+     * Returns the operational domain.
+     *
+     * @return The operational domain.
+     */
+    [[nodiscard]] locked_parallel_flat_hash_map<Key, Value> get_operational_domain() const
     {
         return operational_values;
     }
```

Comment:
This function might be confusing since the associated class is called an `operational_domain` already.

Reply:
I'd suggest `get_operational_values` in accordance with the function below.